File: Digito Verificador/Modulo 10/Teste - Arquivo 10/GerarListaMatriculasMod10.py
def formata_matricula(matricula):
    return f"{matricula:06d}"

def mod10(matricula):
    soma = 0
    for i in range(6):
        digito = int(matricula[i])
        if i % 2 != 0 and digito * 2 > 9:
            soma += digito * 2 - 9
        elif i % 2 != 0 and digito * 2 <= 9:
            soma += digito * 2
        else:
            soma += digito
    dv = soma % 10
    if dv != 0:
        dv = 10 - dv
    return dv

def main():
    try:
        arquivo = open("numeros.txt", "w")

        print("GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM \"DV\"")
        print("Padrao: Modulo 10")
        print("---------------------------------------------------------------")
        print()
        print("Aguarde arquivo sendo gerado... ", end="", flush=True)

        for i in range(1000000):
            matricula = formata_matricula(i)
            dv = mod10(matricula)
            arquivo.write(f"{matricula[:3]}.{matricula[3:]}-{dv}\n")

        arquivo.close()

        print("\n\nSUCESSO: Arquivo gerado.\n")

    except IOError:
        print("Erro ao abrir o arquivo.")
        print("Programa encerrado.")

if __name__ == "__main__":
    main()

File: Digito Verificador/Modulo 10/Teste - Arquivo 10/GerarListaMatriculasMod10.cs
using System;
using System.IO;

class Program
{
    static string FormataMatricula(int MATRICULA)
    {
        return MATRICULA.ToString("D6");
    }

    static int Mod10(string MATRICULA)
    {
        int SOMA = 0;
        for (int I = 0; I <= 5; I++)
        {
            int DIGITO = MATRICULA[I] - '0';
            if (I % 2 != 0 && DIGITO * 2 > 9)
                SOMA += DIGITO * 2 - 9;
            else if (I % 2 != 0 && DIGITO * 2 <= 9)
                SOMA += DIGITO * 2;
            else
                SOMA += DIGITO;
        }
        int DV = (SOMA % 10 == 0) ? 0 : 10 - SOMA % 10;
        return DV;
    }

    static void Main(string[] args)
    {
        StreamWriter ARQUIVO = new StreamWriter("numeros.txt");

        Console.WriteLine("GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM \"DV\"");
        Console.WriteLine("Padrao: Modulo 10");
        Console.WriteLine("---------------------------------------------------------------");
        Console.WriteLine();

        if (ARQUIVO == null)
        {
            Console.Error.WriteLine("Erro ao abrir o ARQUIVO.");
            Console.WriteLine("Programa encerrado.");
            return;
        }

        Console.Write("Aguarde ARQUIVO sendo gerado... ");

        for (int I = 0; I <= 999999; I++)
        {
            string MATRICULA = FormataMatricula(I);
            int DV = Mod10(MATRICULA);
            ARQUIVO.WriteLine($"{MATRICULA.Substring(0, 3)}.{MATRICULA.Substring(3)}-{DV}");
        }

        ARQUIVO.Close();

        Console.WriteLine("\n\nSUCESSO: Arquivo gerado.\n");
    }
}
